modules/sagemaker/sagemaker-templates-service-catalog/templates/hf_import_models/product_stack.py

```diff
@@ -0,0 +1,215 @@
+#
+# SPDX-License-Identifier: MIT-0
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this
+# software and associated documentation files (the "Software"), to deal in the Software
+# without restriction, including without limitation the rights to use, copy, modify,
+# merge, publish, distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+# INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+# PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+# SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+
+import aws_cdk
+import aws_cdk.aws_servicecatalog as servicecatalog
+from aws_cdk import Aws, Tags
+from aws_cdk import aws_iam as iam
+from aws_cdk import aws_kms as kms
+from aws_cdk import aws_s3 as s3
+from aws_cdk import aws_s3_assets as s3_assets
+from aws_cdk import aws_sagemaker as sagemaker
+from constructs import Construct
+
+from templates.hf_import_models.pipeline_constructs.build_pipeline_construct import BuildPipelineConstruct
+
+
+class Product(servicecatalog.ProductStack):
+    DESCRIPTION: str = "Enables the import of Hugging Face models"
+    TEMPLATE_NAME: str = "Hugging Face Model Import"
+
+    def __init__(
+        self,
+        scope: Construct,
+        construct_id: str,
+        build_app_asset: s3_assets.Asset,
+        deploy_app_asset: s3_assets.Asset,
+        **kwargs,
+    ) -> None:
+        super().__init__(scope, construct_id)
+
+        # Define required parmeters
+        project_name = aws_cdk.CfnParameter(
+            self,
+            "SageMakerProjectName",
+            type="String",
+            description="The name of the SageMaker project.",
+            min_length=1,
+            max_length=32,
+        ).value_as_string
+
+        project_id = aws_cdk.CfnParameter(
+            self,
+            "SageMakerProjectId",
+            type="String",
+            min_length=1,
+            max_length=16,
+            description="Service generated Id of the project.",
+        ).value_as_string
+
+        staging_account = aws_cdk.CfnParameter(
+            self,
+            "StgAccountId",
+            type="String",
+            min_length=1,
+            max_length=16,
+            description="Staging account id.",
+        ).value_as_string
+
+        prod_account = aws_cdk.CfnParameter(
+            self,
+            "ProdAccountId",
+            type="String",
+            min_length=1,
+            max_length=16,
+            description="Prod account id.",
+        ).value_as_string
+
+        hf_access_token = aws_cdk.CfnParameter(
```

Comment:
As discussed please change this to `hf_access_token_secret_name` and pull the secret from SSM in the pipeline runtime script. Thank you!